src/lib/mp3/decoder.c

```diff
@@ -2736,8 +2736,8 @@ bool mp3dec000427d8(struct asistream *stream, u32 gr)
 	f32 *sp0c = stream->unk4664[0][0].unk00;
 	s32 i = 0;
 
-	if (stream->window_switching[gr][0] && stream->window_switching[gr][2] == 2) {
-		if (stream->window_switching[gr][4]) {
+	if (stream->window_switching[gr][0] && stream->block_type[gr][0] == 2) {
+		if (stream->mixed_block_flag[gr][0]) {
```

Comment:
Since I'm unfamiliar why this code and the bug being fixed here is not described in the PR, I'm wondering how you discovered that this is the correct fix and if it improves the mp3 decoding in some way?

Reply:
It's an incorrect conversion from the assembly version from what I can tell, but it would typically compile to the same thing.

From `src/lib/mp3/mp3.h`:
```
	/*0x3c98*/ u32 window_switching[2][1];
	/*0x3ca0*/ u32 block_type[2][1];
	/*0x3ca8*/ u32 mixed_block_flag[2][1];
```

Reply:
Oh, I see. It's reaching too far away and into the next array. Thanks for the explanation.

Reply:
if you think there's an incorrect assembly conversion that is also present in the original decomp you should probably also notify @RyanDwyer.

Reply:
Ah right, I'll bear that in mind in future.

It appears that AXDOOMER has already taken that initiative in this case: https://gitlab.com/ryandwyer/perfect-dark/-/merge_requests/6

Reply:
This change has now been pulled into the `port` branch from the upstream repo. Looks like it won't conflict anyway because it's literally the same change.